src/Nbody/Sinks.cpp

```diff
@@ -463,6 +461,29 @@ void Sinks<ndim>::AccreteMassToSinks
     //---------------------------------------------------------------------------------------------
 
 
+    // Determine the closest sinks to all particles that are inside sinks
+#pragma omp for schedule(dynamic,1)
```

Comment:
Yes, but it contains a loop over Nsinks which makes the order of the whole loop Nhydro*Nsinks.  However, looking at it further now,  it's also checking for the particles which are inside sinks, which will only likely be a small fraction.  Then I guess most of the time this loop just cycles over to the next particle so is probably best to change it.  So I guess what you're suggesting is to make a list of particles which have sinkid != -1 and then loop/iterate over that list only?  That sounds like a reasonable solution anyway.